### ISSUE TITLE
[Android] Uploading a file takes 10 times as long as the website

### ISSUE THREAD

**Issue by Softvision-AndreeaCupsa** [mozilla/send#1139]
**[Notes]:**
- Uploading a file using the Firefox app on the phone and the “https://send2.dev.lcip.org/” website takes about as much time as the website on the computer.
- Upload taking a long time might be an issue because if the screen auto locks, and remains locked for 1 minute, the upload stops as all background applications are stopped.

**[Affected Versions]:**
- Android 8.0 (Samsung S8, Google Pixel, Huawei P20Lite)

**[Prerequisites]:**
- Have the latest Send Android build (org.mozilla.sendandroid-2019-02-06-signed.apk) installed.
- Have a file of 100MB size saved on the device. 

**[Steps to reproduce]:**
1. Open the Send Android application and add a 100MB file.
2. Open the browser, navigate to https://send2.dev.lcip.org/ and add a 50MB file.
3. Start uploading both files at the same time.
4. Observe the mobile app’s behavior as the website upload has finished.

**[Expected result]:**
- The mobile upload has finished or is almost finished.

**[Actual result]:**
- The mobile upload is still under 15%.

**[Notes]:**
- Upload speed on Samsung S8 (wifi connection): 84 Mbps.
- Pc upload speed 15.59 Mbps.
- Uploading a file of 150MB took ~20 minutes.
- Attached a screen recording of the issue: 

![rec of issue - upload](https://user-images.githubusercontent.com/37875464/52415388-86491e80-2aef-11e9-9c1c-3b3d5316e216.gif)

**Comment by dannycoates** [comment]
@fzzzy the base64 encoding only applies to intent triggered files, right? So if it's also slow for files added from the 'add files' button then the slowness is all within the webview.

**Comment by fzzzy** [comment]
@dannycoates good point! my main guess at the moment is that the cryptofill is being used because the webview doesn't have some api, or because the user agent is set to "send android"

**Comment by dannycoates** [comment]
android can't be loading the crypto polyfill because it never calls `getCapabilities()`

**Comment by SoftVision-CosminMuntean** [comment]
I have retested this issue using the latest Android Send version (org.mozilla.sendandroid-2019-03-01-release-signed.apk) compared with the Send v2 website by uploading a **100 Mb file**. Unfortunately I haven't observed any noticeable improvements. Here are the results:

- On Send v2 websites took: ~20 seconds
- On Samsung S8 (Android 8.0.0) took: ~ 2m and 10s
- On Google Pixel (Android 8.0.0) took: ~ 3m and 30s

**Comment by fzzzy** [comment]
@SoftVision-CosminMuntean Are you testing uploading with the Chrome browser on the same android device you are testing the app on? Because I am not seeing any of that behavior on my S8.

**Comment by SoftVision-CosminMuntean** [comment]
@fzzzy I have tested with Firefox 65.0.1 browser on the same Android device where the app is installed.

However, I just verified on Chrome browser on the same Android device and it seems that 100Mb file takes ~3m10s to be uploaded and on Firefox browser only ~20s.

**Comment by fzzzy** [comment]
@SoftVision-CosminMuntean Ok, great, that is very good news. That means switching to GeckoView will most likely fix this. However, there are no engineers on the project at the moment, as I am starting on rust-components today. If Send becomes a priority again and engineers are assigned to the project, there is a clear path to speeding it up.
